corehq/apps/users/tests/test_db_accessors.py

```diff
@@ -91,3 +92,9 @@ def test_get_all_ids(self):
     def test_get_id_by_username(self):
         user_id = get_user_id_by_username(self.ccuser_1.username)
         self.assertEqual(user_id, self.ccuser_1._id)
+
+    def test_get_deleted_by_username(self):
+        self.ccuser_1.retire()
+        self.assertEqual(self.ccuser_1.username, get_deleted_by_username(CommCareUser,
+                                                                         self.ccuser_1.username).username)
```

Comment:
also very minor but checking the `._id` instead of `.username` might be a better confirmation that you got the same thing back